Document PreStop hook and it's env vars (#2274)
elastic/cloud-on-k8s: docs/elasticsearch-specification.asciidoc

> [source,yaml]

Reviewer: I think this needs to be `[source,yaml,subs="attributes"]` to sub the attr, see here:
http://cloud-on-k8s_2274.docs-preview.app.elstc.co/guide/en/cloud-on-k8s/master/k8s-prestop.html

Author: Nice, thanks!

----

Note that this will require restarting the pods.

[id="{p}-prestop"]
=== Pod PreStop hook

> When Elasticsearch `Pod` is terminated, its `Endpoint` is removed from the `Service` and the Elasticsearch process is terminated. As these two operations happen in parallel, a race condition exists. If the Elasticsearch process is already shut down, but the `Endpoint` is still a part of the `Service`, any new connection might fail. For more information, see link:https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods[Termination of pods].

Reviewer: ```suggestion
When an Elasticsearch `Pod` is terminated, its `Endpoint` is removed from the `Service` and the Elasticsearch process is terminated. As these two operations happen in parallel, a race condition exists. If the Elasticsearch process is already shut down, but the `Endpoint` is still a part of the `Service`, any new connection might fail. For more information, see link:https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods[Termination of pods].
```

> When Elasticsearch `Pod` is terminated, it's `Endpoint` is being removed from the `Service` and the Elasticsearch process is terminated. As these two operations are happening in parallel, a race condition exists. If the Elasticsearch process is already shut down, but the `Endpoint` is still being a part of the `Service`, any new connection might fail. For more information, see link:https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods[Termination of pods].

Reviewer 2: ```suggestion
When Elasticsearch `Pod` is terminated, its `Endpoint` is removed from the `Service` and the Elasticsearch process is terminated. As these two operations happen in parallel, a race condition exists. If the Elasticsearch process is already shut down, but the `Endpoint` is still a part of the `Service`, any new connection might fail. For more information, see link:https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods[Termination of pods].
```

> To address this issue and guarantee no downtime for new connections, ECK relies on PreStop lifecycle hook. This allows to delay Elasticsearch process termination until `Service` DNS record won't contain the IP of the `Pod`. First, PreStop lifecycle hook will keep querying DNS for `MAX_WAIT_SECONDS` (defaulting to 20) and then it will wait for `ADDITIONAL_WAIT_SECONDS` (defaulting to 1). Additional wait is needed to give clients time to use IP resolved just before DNS record was updated. The exact behavior is configurable using environment variables, for example:

Reviewer: ```suggestion
To address this issue and guarantee no downtime for new connections, ECK relies on a link:https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/[PreStop lifecycle hook]. This waits to terminate the Elasticsearch process until the `Service` DNS record does not contain the IP of the `Pod`. First, the PreStop lifecycle hook will keep querying DNS for `MAX_WAIT_SECONDS` (defaulting to 20) and then it will wait for `ADDITIONAL_WAIT_SECONDS` (defaulting to 1). Additional wait is used to give clients time to use an IP resolved just before DNS record was updated. The exact behavior is configurable using environment variables, for example:
```

> [source,yaml]

Reviewer: Same here with the attr flag

> To address this issue and guarantee no downtime for new connections, ECK relies on PreStop lifecycle hook. This allows to delay Elasticsearch process termination until `Service` DNS record won't contain the IP of the `Pod`. First, PreStop lifecycle hook will keep querying DNS for `MAX_WAIT_SECONDS` (defaulting to 20) and then it will wait for `ADDITIONAL_WAIT_SECONDS` (defaulting to 1). Additional wait is needed to give clients time to use IP resolved just before DNS record was updated. The exact behavior is configurable using environment variables, for example:

Reviewer 2: ```suggestion
To address this issue and guarantee no downtime for new connections, ECK relies on PreStop lifecycle hook. This allows you to delay Elasticsearch process termination until the `Service` DNS record contains the IP of the `Pod`. First, PreStop lifecycle hook keeps querying DNS for `MAX_WAIT_SECONDS` (defaulting to 20). Then, it waits for `ADDITIONAL_WAIT_SECONDS` (defaulting to 1). Additional time is needed so that clients can use the IP resolved just before DNS record gets updated. To configure the exact behavior, use environment variables, as in this example:
```

Author: Before:
> This allows to delay Elasticsearch process termination until `Service` DNS record won't contain the IP of the `Pod`.

After:
> This allows you to delay Elasticsearch process termination until the `Service` DNS record contains the IP of the `Pod`.

The message changed here :) would below be fine?

This allows you to delay Elasticsearch process termination until the `Service` DNS record doesn't contain the IP of the `Pod`.
